dds/DCPS/XTypes/DynamicData.h

```diff
@@ -11,26 +11,29 @@
 
 #include <dds/DCPS/Serializer.h>
 #include <dds/DCPS/PoolAllocator.h>
-#include <dds/CorbaSeq/LongSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/ULongSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/Int8SeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/UInt8SeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/ShortSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/UShortSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/LongLongSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/ULongLongSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/FloatSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/DoubleSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/LongDoubleSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/CharSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/WCharSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/OctetSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/BooleanSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/StringSeqTypeSupportImpl.h>
-#include <dds/CorbaSeq/WStringSeqTypeSupportImpl.h>
 
 OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
 
+namespace CORBA {
+class LongSeq;
+class ULongSeq;
+class Int8Seq;
+class UInt8Seq;
+class ShortSeq;
+class UShortSeq;
+class LongLongSeq;
+class ULongLongSeq;
+class FloatSeq;
+class DoubleSeq;
+class LongDoubleSeq;
+class CharSeq;
+class WCharSeq;
+class OctetSeq;
+class BooleanSeq;
+class StringSeq;
+class WStringSeq;
+}
+
```

Comment:
OK, this seems like it's the namespace conflict when versioned namespaces are enabled.  So perhaps my other comment isn't relevant (but I'm going to leave it in the review just in case).

If the forward declaration of these `*Seq` types is needed, they don't go within `OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL` because they belong to TAO.  So they would need the corresponding TAO macro.  However, I don't think that's the way to go.  Probably just include the headers you need for the sequence types, like `tao/FloatSeqC.h` for the sequence of floats, and so on for all built-in types.

Reply:
There is a circular include problem due to ServiceParticipant.h including Recorder.h which includes DynamicData.h which includes the sequence include files which each include ServiceParticipant.h. This is why I forward declared these here.

Do you have a suggestion on how to solve this issue?

Reply:
Note that my comment yesterday was about `tao/FloatSeqC.h`, not an OpenDDS header.